src/libsyntax_ext/deriving/cmp/partial_eq.rs

```diff
@@ -52,7 +55,10 @@ pub fn expand_deriving_partial_eq(cx: &mut ExtCtxt,
                     _ => cx.span_bug(span, "not exactly 2 arguments in `derive(PartialEq)`")
                 };
 
-                let eq = cx.expr_binary(span, BinOpKind::Ne, self_f, other_f.clone());
+                let eq = cx.expr_binary(span,
+                                        BinOpKind::Ne,
+                                        cx.expr_addr_of(span, self_f),
+                                        cx.expr_addr_of(span, other_f.clone()));
```

Comment:
Yeah maybe we should use UFCS in the derive expansion, but there's a deeper issue (which is why @bluss suggested not working around it here)... line 17 should be exactly the same as line 19, right?